crates/autopilot/src/database/onchain_order_events.rs

```diff
@@ -197,9 +197,15 @@ impl<T: Sync + Send + Clone, W: Sync + Send + Clone> EventStoring<ContractEvent>
         .await
         .context("append_onchain_orders failed")?;
 
-        insert_quotes(&mut transaction, quotes.as_slice())
-            .await
-            .context("appending quotes for onchain orders failed")?;
+        // It's okay to not store quotes for orders is with placement_errors as
+        // only orders that will be used in auctions need a quote stored
+        // to calcualte the rewards.
```

Comment:
Nit, I find the comment a bit confusing. Note that this comment is duplicated and also the other should be changed, in case!

```suggestion
        // We only need to insert quotes for orders that will be included in an
        // auction (they are needed to compute solver rewards). If placement
        // failed, then the quote is not needed.
```